Silicon/Qualcomm/sm8250/Library/PlatformMemoryMapLib/PlatformMemoryMapLib.c

```diff
@@ -0,0 +1,62 @@
+
+static ARM_MEMORY_REGION_DESCRIPTOR_EX gDeviceMemoryDescriptorEx[] = {
+	{"Hypervisor",       0x80000000, 0x000600000, AddMem, SYS_MEM, SYS_MEM_CAP, Reserv, NS_DEVICE},
+	{"HLOS",             0x80600000, 0x000100000, AddMem, SYS_MEM, SYS_MEM_CAP, RtCode, WRITE_BACK_XN},
+	{"AOP",              0x80700000, 0x000160000, AddMem, MEM_RES, UNCACHEABLE, Reserv, UNCACHED_UNBUFFERED_XN},
+	{"AOP CMD DB",       0x80860000, 0x000020000, AddMem, MEM_RES, UNCACHEABLE, Reserv, UNCACHED_UNBUFFERED_XN},
+	{"XBL Log Buffer",   0x80880000, 0x000014000, AddMem, SYS_MEM, SYS_MEM_CAP, Reserv, WRITE_BACK_XN},
+	{"HLOS",             0x80894000, 0x00006C000, AddMem, SYS_MEM, SYS_MEM_CAP, Conv,   WRITE_BACK_XN},
+	{"SMEM",             0x80900000, 0x000200000, AddMem, MEM_RES, UNCACHEABLE, Reserv, UNCACHED_UNBUFFERED_XN},
+	{"TZ",               0x80b00000, 0x005700000, AddMem, SYS_MEM, SYS_MEM_CAP, Reserv, NS_DEVICE},
+	{"PIL Reserved",     0x86200000, 0x005D00000, AddMem, SYS_MEM, SYS_MEM_CAP, Reserv, UNCACHED_UNBUFFERED_XN},
+	{"CDSP Secure Heap", 0x8BF00000, 0x004600000, AddMem, SYS_MEM, SYS_MEM_CAP, Reserv, UNCACHED_UNBUFFERED_XN},
+	{"HLOS",             0x90500000, 0x00BB00000, AddMem, SYS_MEM, SYS_MEM_CAP, BsData, WRITE_BACK_XN},
+	{"Display Reserved", 0x9C000000, 0x002400000, AddMem, MEM_RES, SYS_MEM_CAP, Reserv, WRITE_THROUGH_XN},
+	{"DBI Dump",         0x9E400000, 0x000F00000, NoHob,  MMAP_IO, INITIALIZED, Conv,   UNCACHED_UNBUFFERED_XN},
+	{"HLOS",             0x9F300000, 0x000C00000, AddMem, SYS_MEM, SYS_MEM_CAP, Conv,   WRITE_BACK_XN},
+	{"SEC Heap",         0x9FF00000, 0x00008C000, AddMem, SYS_MEM, SYS_MEM_CAP, BsData, WRITE_BACK_XN},
+	{"CPU Vectors",      0x9FF8C000, 0x000001000, AddMem, SYS_MEM, SYS_MEM_CAP, BsData, WRITE_BACK},
+	{"MMU PageTables",   0x9FF8D000, 0x000003000, AddMem, SYS_MEM, SYS_MEM_CAP, BsData, WRITE_BACK_XN},
+	{"UEFI Stack",       0x9FF90000, 0x000040000, AddMem, SYS_MEM, SYS_MEM_CAP, BsData, WRITE_BACK_XN},
+	{"HLOS",             0x9FFD0000, 0x000027000, AddMem, SYS_MEM, SYS_MEM_CAP, Conv,   WRITE_BACK_XN},
+	{"Log Buffer",       0x9FFF7000, 0x000008000, AddMem, SYS_MEM, SYS_MEM_CAP, Reserv, WRITE_BACK_XN},
+	{"Info Blk",         0x9FFFF000, 0x000001000, AddMem, SYS_MEM, SYS_MEM_CAP, Reserv, WRITE_BACK_XN},
+	{"DXE Heap",         0xA0000000, 0x007900000, AddMem, SYS_MEM, SYS_MEM_CAP, Conv,   WRITE_BACK_XN},
+	{"HLOS",             0xA7900000, 0x008700000, AddMem, SYS_MEM, SYS_MEM_CAP, Conv,   WRITE_BACK_XN},
+	{"PStore",           0xB0000000, 0x010000000, AddMem, SYS_MEM, SYS_MEM_CAP, Reserv, WRITE_BACK_XN},
+	{"HLOS",             0xB0080000, 0x00CB80000, AddMem, SYS_MEM, SYS_MEM_CAP, Conv,   WRITE_BACK_XN},
+	{"HLOS",             0xC0000000, 0x010000000, AddMem, SYS_MEM, SYS_MEM_CAP, Conv,   WRITE_BACK_XN},
+	{"UEFI FD",          0xD0000000, 0x000600000, AddMem, SYS_MEM, SYS_MEM_CAP, BsData, WRITE_BACK},
+	{"HLOS",             0xD0600000, 0x12FA00000, AddMem, SYS_MEM, SYS_MEM_CAP, Conv,   WRITE_BACK_XN},
```

Comment:
Where is the rest of ram? Without it no OS will really have anything to work with. Also nit: using Add6GMem etc maybe?